src/stream/src/executor/subscription.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::time::Duration;
+
+use futures::prelude::stream::StreamExt;
+use futures_async_stream::try_stream;
+use risingwave_common::catalog::Schema;
+use risingwave_common::types::Timestamptz;
+use risingwave_common::util::epoch::Epoch;
+use risingwave_storage::store::LocalStateStore;
+use tokio::time::Instant;
+
+use super::{
+    expect_first_barrier, ActorContextRef, BoxedExecutor, BoxedMessageStream, Executor,
+    ExecutorInfo, Message, PkIndicesRef, StreamExecutorError, StreamExecutorResult,
+};
+use crate::common::log_store_impl::kv_log_store::ReaderTruncationOffsetType;
+use crate::common::log_store_impl::subscription_log_store::SubscriptionLogStoreWriter;
+
+const EXECUTE_GC_INTERVAL: u64 = 3600;
+
+pub struct SubscriptionExecutor<LS: LocalStateStore> {
+    actor_context: ActorContextRef,
+    info: ExecutorInfo,
+    input: BoxedExecutor,
+    log_store: SubscriptionLogStoreWriter<LS>,
+    retention: i64,
```

Comment:
Better to rename to `retention_seconds` to avoid filling duration of other unit by mistake.

Reply:
So as the field name in proto.